src/components/file-input/file-input.tsx

```diff
@@ -0,0 +1,81 @@
+import './file-input.scss';
+import React, { useState } from 'react';
+import { noop } from '../../helpers/func';
+import { Button } from '../button/button';
+import { v4 as uuidv4 } from 'uuid';
+import { readFile } from './file-reader';
+import { LoadingIcon } from '../../assets/icons/loading-icon/loading-icon';
+import { CancelIcon } from '../../assets/icons/cancel-icon/cancel-icon';
+import { DocIcon } from '../../assets/icons/doc-icon/doc-icon';
+
+export interface ImportedFile {
+  name: string;
+  content: string;
+}
+
+interface FileInputProps {
+  className?: string;
+  label: string;
+  importedFile?: ImportedFile;
+  onImportedFileChange: (importedFile?: ImportedFile) => void;
+}
+
+export const FileInput = ({
+  className = '',
+  label,
+  importedFile,
+  onImportedFileChange,
+}: FileInputProps) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const id = uuidv4();
+  return (
+    <div className={`c-file-input ${className}`}>
+      {importedFile ? getSelectedFile() : getSelectFileButton()}
+      <input key={id} type="file" id={id} name="myfile" onChange={importFile} />
```

Comment:
there's also an `accept` attribute that's supported in some browsers if we want to allow users to only pick specific files

Reply:
good catch. I added it; it's just .txt files (for now, we could always add more if we get bored)